resources/guidelines/testing/e2e-best-practises.md

```diff
@@ -1,59 +1,101 @@
+## Amount and prioritisation of end-to-end tests
 
-* Use risk-analysis: Cover those workflows with E2E tests, which are most vulnerable and would cause most damage 
+### Video
 
-  if broken. 
+When it comes to dividing test types, selecting and prioritizing test cases, and thus designing tests, things get a bit more complicated. We have generally aligned our test strategy with the test pyramid, although not 100%. The pyramid states that end-to-end tests should be written in a few, but well-chosen test cases - because end-to-end tests are slow and therefore expensive.
 
-* Avoid duplicate coverage: E2E tests should only cover what they can cover, usually big-picture user 
+At [Shopware Community Day](https://scd.shopware.com/en/) 2020, we gave a talk on how we approach automated testing in Shopware, how far we have come on this journey, and what we have gained so far:
 
-  stories \(workflows\) that contain many components and views. 
+<!-- markdown-link-check-disable-next-line -->
+{% embed url="https://www.youtube.com/watch?v=sxvQoWF4KS0" %}
 
-* Sometimes unit tests are suited better: For example, use an E2E test to test your application's reaction on a 
+To sum it up briefly for your End-to-End testing context, End-To-End tests are the one tests being slow and thus expensive to maintain. That's why we need a way to prioritize our test cases.
 
-  failed validation, not the validation itself.
+### When should I write an end-to-end test?
 
-## Structure and Scope
+{% hint style="danger" %} Cover every possible workflow with E2E tests. {% endhint %}
 
-The second most important thing is to just test the workflow you explicitly want to test: Any other steps or workflows to get your test running should be done using API operations in the `beforeEach` hook, as we don't want to test them more than once.
+{% hint style="success" %} Use proper prioritisation to choose test cases covered by E2E tests. {% endhint %}
+
+You see, due to running times it is not advisable to cover every single workflow available. The following criteria may help you with that:
+* **Cover the most general, most used workflows of a feature**, e.g. CRUD operations. The term "[happy path](https://en.wikipedia.org/wiki/Happy_path)" describes those workflows quite well.
+* **Beware the critical path**: Cover those workflows with E2E tests, which are most vulnerable and would cause most damage if broken.
+* **Avoid duplicate coverage**: E2E tests should only cover what they can cover, usually big-picture user stories \(workflows\) that contain many components and views.
+  * Sometimes unit tests are suited better: For example, use an E2E test to test your application's reaction on a failed validation, not the validation itself.
+
+## Workflow-based end-to-end tests
+
+{% hint style="danger" %} Write E2E test as you would write unit tests. {% endhint %}
+
+{% hint style="success" %} Write E2E tests in a "workflow-based" manner, that means writing the test describing a reals
```

Comment:
You got me puzzled here:

I assume 
reals -> real

Reply:
Ah, bad typo 🙈